src/main/java/com/amazonaws/kinesisvideo/internal/producer/StreamEventMetadata.java

```diff
@@ -0,0 +1,60 @@
+package com.amazonaws.kinesisvideo.internal.producer;
+
+import javax.annotation.Nullable;
+
+/**
+ * Kinesis Video stream event metadata as provided by the native codebase.
+ */
+public class StreamEventMetadata {
+
+    // Current class version as described by the native struct.
+    public static final int STREAM_EVENT_METADATA_CURRENT_VERSION = 0;
+
+
+    // Version of the struct.
+    int mVersion;
+
+    // Optional s3 prefix.
+    @Nullable
+    String mImagePrefix;
+
+    // Optional optimization stating how many name/value pairs to be appended.
+    // NOTE: This is NOT optional if using mNames and mValues, else the arrays will not be used.
+    @Nullable
+    byte mNumberOfPairs;
+
+    // Optional custom data name/value pairs. String lengths must be <= MKV_MAX_TAG_NAME_LEN as defined in the native code.
+    @Nullable
```

Comment:
Are you sure this is nullable? There's a null check in the API implementation in Producer C which would throw `STATUS_NULL_ARG` if this was null

Reply:
I'm not seeing any null check in Producer C SDK, and the API implementation I believe is in PIC which also does not have such a check, but that is because the pointers to the names and values arrays are initialized with fixed lengths on the struct, so there's no risk of them being null.

My thinking is for code-parity, we might want to allow passing no arrays into the PIC layer since this is technically supported if constructing the metadata struct using designated initialization as we do in the PIC unit tests:
`StreamEventMetadata Meta{STREAM_EVENT_METADATA_CURRENT_VERSION, NULL, 1, {}, {}}`. And from my testing, calling putKinesisEventMetadata with the following has no errors:
`StreamEventMetadata Meta{STREAM_EVENT_METADATA_CURRENT_VERSION, NULL, 1, NULL, NULL}`
(the last 2 members are the names and values arrays).